app/models/merchant.rb

```diff
@@ -14,4 +15,18 @@ def single_merchant_revenue
       .joins(:transactions, :invoice_items)
       .where(transactions: {result: "Success"})[0]
   end
+
+  def customers_with_pending_invoices
+    Customer.find_by_sql ["SELECT customers.* FROM merchants
+                          JOIN invoices ON merchants.id = invoices.merchant_id
+                          JOIN customers ON customers.id = invoices.customer_id
+                          JOIN transactions ON invoices.id = transactions.invoice_id WHERE merchants.id = #{id}
+                          EXCEPT
+                          SELECT customers.* FROM merchants
+                          JOIN invoices ON merchants.id = invoices.merchant_id
+                          JOIN customers ON customers.id = invoices.customer_id
+                          JOIN transactions ON invoices.id = transactions.invoice_id
+                          WHERE merchants.id = #{id} and transactions.result = 'success';"]
+
```

Comment:
Just curious on this query - What's the need for using the except to do multiple selects?

Reply:
The EXCEPT clause returns whichever records appear in the first select query that do not appear in the second select query. The first query returns all customers of a given merchant, and the second returns only those customers with a successful transaction result. The EXCEPT clause then produces all customers of a given merchant that have NOT had a successful transaction, i.e. customers with pending transactions.

Reply:
Interesting!